src/EnergyPlus/OutputReports.cc

```diff
@@ -2575,134 +1977,79 @@ void VRMLOut(std::string &PolygonAction, std::string &ColorScheme)
-                    ObjexxFCL::gio::write(csidenumber, fmtLD) << vv1 - 1;
-                    strip(csidenumber);
-                    {
-                        IOFlags flags;
-                        flags.ADVANCE("No");
-                        ObjexxFCL::gio::write(unit, Format_804, flags) << ' ' + csidenumber;
-                    }
-                    vv2 = mytriangles(svert).vv2;
-                    ObjexxFCL::gio::write(csidenumber, fmtLD) << vv2 - 1;
-                    strip(csidenumber);
-                    {
-                        IOFlags flags;
-                        flags.ADVANCE("No");
-                        ObjexxFCL::gio::write(unit, Format_804, flags) << ' ' + csidenumber;
-                    }
-                    ObjexxFCL::gio::write(unit, Format_804) << " -1";
+                Array1D<dTriangle> mytriangles;
+                const auto ntri = Triangulate(Surface(surf).Sides,
+                                              Surface(surf).Vertex,
+                                              mytriangles,
+                                              Surface(surf).Azimuth,
+                                              Surface(surf).Tilt,
+                                              Surface(surf).Name,
+                                              Surface(surf).Class);
+                for (int svert = 1; svert <= ntri; ++svert) {
+                    const auto vv0 = mytriangles(svert).vv0;
+                    const auto vv1 = mytriangles(svert).vv1;
+                    const auto vv2 = mytriangles(svert).vv2;
+                    print(wrlfile, " {} {} {} -1\n", vv0 - 1, vv1 - 1, vv2 - 1);
                 }
-                ObjexxFCL::gio::write(unit, Format_805);
+                print(wrlfile, Format_805);
                 mytriangles.deallocate();
             }
         }
         // still have to do shading surfaces for zone
         colorindex = 4;
-        for (surf = 1; surf <= TotSurfaces; ++surf) {
+        for (int surf = 1; surf <= TotSurfaces; ++surf) {
             //      !if (surface(surf)%heattranssurf) CYCLE ! Shading with a construction is allowed to be HT surf for daylighting shelves
             if (Surface(surf).Class != SurfaceClass_Shading) continue;
             if (Surface(surf).ZoneName != Zone(zones).Name) continue;
             if (Surface(surf).Sides == 0) continue;
-            ObjexxFCL::gio::write(unit, Format_710) << "# " + Surface(surf).ZoneName + ':' + Surface(surf).Name;
-            ObjexxFCL::gio::write(unit, Format_801) << colorstring(colorindex) << "Surf" + csurfnumber;
-            for (vert = 1; vert <= Surface(surf).Sides; ++vert) {
-                ObjexxFCL::gio::write(unit, Format_802) << Surface(surf).Vertex(vert).x << Surface(surf).Vertex(vert).y << Surface(surf).Vertex(vert).z;
+            print(wrlfile, "# {}:{}\n", Surface(surf).ZoneName, Surface(surf).Name);
+            print(wrlfile, Format_801, colorstring(colorindex), "Surf", surf);
```

Comment:
Here the bug in https://github.com/NREL/EnergyPlus/pull/7972/files#r423837070 is fixed because the variable `csurfnumber` was removed from the top level of the function, so it was impossible to accidentally reuse the value here.